tests/unit/serialization/polymorphic/serialization_polymorphic_nonintrusive_abstract.cpp

```diff
@@ -0,0 +1,143 @@
+//  Copyright (c) 2015 Thomas Heller
+//  Copyright (c) 2015 Anton Bikineev
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/runtime/serialization/serialize.hpp>
+#include <hpx/runtime/serialization/base_object.hpp>
+
+#include <hpx/runtime/serialization/input_archive.hpp>
+#include <hpx/runtime/serialization/output_archive.hpp>
+
+#include <hpx/util/lightweight_test.hpp>
+
+template <typename T>
+struct Base
+{
+    Base()
+    {}
+
+    Base(std::string prefix) : prefix_(prefix)
+    {}
+
+    virtual ~Base()
+    {}
+
+    virtual std::size_t size() = 0;
+    virtual std::string print() = 0;
+
+    std::string prefix_;
+};
+
+namespace hpx { namespace traits {
+    template <typename T>
+    struct is_nonintrusive_polymorphic<Base<T> >
+      : boost::mpl::true_ {};
+}}
+
+template <typename Archive, typename T>
+void serialize(Archive & ar, Base<T> & b, unsigned)
+{
+    ar & b.prefix_;
+}
+
+template <typename T>
+struct Derived1 : Base<T>
+{
+    Derived1() {}
+
+    Derived1(std::string prefix, std::size_t size)
+      : Base<T>(prefix)
+      , size_(size)
+    {}
+
+    std::size_t size()
+    {
+        return size_;
+    }
+
+    std::size_t size_;
+};
+
+namespace hpx { namespace traits {
+    template <typename T>
+    struct is_nonintrusive_polymorphic<Derived1<T> >
+      : boost::mpl::true_ {};
+}}
+
```

Comment:
Yes, unfortunately, the corresponding macro for templated types hasn't been implemented.